### PR TITLE
[#3700] support the rendering of whatsapp cloud messages

### LINKED ISSUE #3700: Support the rendering of WhatsApp Cloud messages
**Current state:** We're already supporting `WhatsApp Business Cloud` both in the frontend and backend, but we're only rendering text messages at the moment.

**To do:** render all inbound messages sent via the `WhatsApp Business Cloud` connector.

**Ressources:** 
https://developers.facebook.com/docs/whatsapp/cloud-api/reference/

### PR DESCRIPTION
**Pull request by AudreyKj**
closes #3700 

**Changes:** 
-  improvements to the render library (renamed Facebook to Meta for better clarity + fixes)
 - WhatsAppMedia
 - WhatsAppLocation
 - WhatsAppInteractive
 - WhatsAppContacts

reference: https://developers.facebook.com/docs/whatsapp/cloud-api/reference

Screenshot 2022-09-14 at 17 40 29

### CODE REVIEW

File: lib/typescript/render/components/CurrentLocation/index.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        Latitude: {latitude}, Longitude: {longitude}
+      </p>
+      {name && (
+        <p className={styles.geolocation}>
+          {' '}
+          {name}{' '}
+          {address && (
+            <a target="_blank" rel="noopener noreferrer" href={`http://maps.google.com/?q='${address}`}>
+              {address}
+            </a>
```
Comment (AudreyKj) [inline lib/typescript/render/components/CurrentLocation/index.tsx:33]: improved Location component with a link to Google Maps

File: lib/typescript/render/components/Image/index.tsx [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
 import {ImageWithFallback} from '../ImageWithFallback';
-import {ImageContent} from '../../providers/facebook/facebookModel';
+import {ImageContent} from '../../providers/meta/MetaModel';
 import {Text} from '../../components/Text';
```
Comment (AudreyKj) [inline lib/typescript/render/components/Image/index.tsx:4]: renamed the` Facebook `provider to `Meta` in the render lib

File: lib/typescript/render/providers/meta/MetaRender.tsx [hunk 3]
```diff
@@ -174,11 +210,11 @@ const parseAttachment = (
 
   return {
     type: 'text',
-    text: 'Unsupported message type',
+    text: JSON.stringify(attachment),
   };
```
Comment (AudreyKj) [inline lib/typescript/render/providers/meta/MetaRender.tsx:214]: we are displaying the payload if the message is not recognized

File: lib/typescript/render/providers/meta/MetaModel.ts [hunk 4]
```diff
@@ -197,7 +306,12 @@ export type ContentUnion =
   | StoryRepliesContent
   | ShareContent
   | Fallback
-  | DeletedMessageContent;
+  | DeletedMessageContent
+  | WhatsAppTemplate
+  | WhatsAppMedia
+  | WhatsAppLocation
+  | WhatsAppInteractive
+  | WhatsAppContacts;
```
Comment (AudreyKj) [inline lib/typescript/render/providers/meta/MetaModel.ts:315]: WhatsApp messages types

File: frontend/inbox/src/pages/Inbox/MessageInput/index.tsx [hunk 5]
```diff
@@ -283,6 +283,9 @@ const MessageInput = (props: Props) => {
         case Source.viber:
           message.message = outboundMapper.getTextPayload(input);
           break;
+        case Source.whatsapp:
+          message.message = outboundMapper.getTextPayload(input);
+          break;
       }
```
Comment (AudreyKj) [inline frontend/inbox/src/pages/Inbox/MessageInput/index.tsx:289]: outbound mapper for whatsapp: right now we supporting only text

**Review by thorstenairy — APPROVED** [review]

**Pull request merged**